src/glum/_glm.py

```diff
@@ -2724,7 +2734,8 @@ def _set_up_and_check_fit_args(
 
                 self.feature_dtypes_ = X.dtypes.to_dict()
                 self.has_missing_category_ = {
-                    col: (self.cat_missing_method == "convert") and X[col].isna().any()
+                    col: (getattr(self, "cat_missing_method", "fail") == "convert")
+                    and X[col].isna().any()
                     for col, dtype in self.feature_dtypes_.items()
                     if isinstance(dtype, pd.CategoricalDtype)
                 }
```

Comment:
Note that, because `self.has_missing_category_` is created here, we don't need to use `getattr(self, "has_missing_category_", ...)` in 2789 and 2792 anymore.